adafruit_lis3dh/lis3dh.py

```diff
@@ -113,6 +115,31 @@ def acceleration(self):
 
         return (x / divider * 9.806, y / divider * 9.806, z / divider * 9.806)
 
+    def shake(self, shake_threshold=30, avg_delay=0.01, avg_count=10):
+        """Detect when the accelerometer is shaken. This takes a series of readings
+        from acceleration, finds the average, then calculates the total acceleration
+        by taking the square root of the sum of squares, i.e. sqrt(X*X + Y*Y + Z*Z).
+        The result is compared to a threshold to determine if the device has been
+        shaken.
+        Optional parameters:
+         shake_threshold - Increase or decrease to change shake sensitivity. This
+                           requires a minimum value of 10. 10 is the total
+                           acceleration if the board is not moving, therefore
+                           anything less than 10 will erroneously report a constant
+                           shake detected. (Default 30)
+         avg_delay - The speed in seconds at which the series of values used for the
+                     average acceleration are taken. (Default 0.01)
```

Comment:
I think it'd be clearer if this was total delay. That way its easy to tell how long it will take. Then, in the sleep just divide total delay by the number of samples.